src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -902,6 +902,38 @@ class RBDService : public ServiceBase {
     }
 };
 
+class WNBDWatchCtx : public librbd::UpdateWatchCtx
+{
+private:
+  librados::IoCtx &io_ctx;
+  WnbdHandler* handler;
+  librbd::Image &image;
+  uint64_t size;
+public:
+  WNBDWatchCtx(librados::IoCtx& io_ctx, WnbdHandler* handler,
+               librbd::Image& image, uint64_t size)
+    : io_ctx(io_ctx)
+    , handler(handler)
+    , image(image)
+    , size(size)
+  { }
+
+  ~WNBDWatchCtx() override {}
+
+  void handle_notify() override
+  {
+    librbd::image_info_t info;
+    if (image.stat(info, sizeof(info)) == 0) {
+      uint64_t new_size = info.size;
```

Comment:
Since you are introducing an unnecessary variable (presumably to stress that `info.size` is the "new" size), you might want to call `image.size` instead of `image.stat` and get rid of `librbd::image_info_t` altogether.